app/views/comments/_form.html.haml

```diff
@@ -1,23 +1,22 @@
-= render :partial => 'comments/help'
-
+= render :partial => 'comments/help', locals: { id: id}
 = form_for [ parent, comment ], html: {role: "form"} do |f|
   %p
     .comment-form-heading
       %ul.nav.nav-tabs
-        %li.active#show-source{ role: :presentation }
-          %a{ href: '#markdown-source', role: :tab, data: { toggle: :tab } } Edit
-        %li#show-preview{ role: :presentation }
-          %a{ href: '#markdown-preview', role: :tab, data: { toggle: :tab } } Preview
+        %li.active{ id: "show-source#{id}", role: :presentation }
```

Comment:
Why do you need all this `ids`, maybe because you are using `id` instead of `class`? if `show-source`, `show-preview` was a class you shouldn't need the random `id`. Even in the [js](https://github.com/SUSE/hackweek/pull/237/files#diff-a9c3bd311eab80c9ebe6a69830f9ad02R55) you are no using the `ids` you go to the form and after to the element you need to toggle.